insights/tests/client/connection/test_checkin.py

```diff
@@ -33,7 +34,7 @@ def _get_canonical_facts_response(canonical_facts):
     **{"return_value.status_code": codes.CREATED}
 )
 @patch("insights.client.connection.InsightsConnection.get_proxies")
-def test_canonical_facts_request(get_proxies, post, get_canonical_facts, rm_conf):
+def test_canonical_facts_request(get_proxies, post, get_canonical_facts, fetch_system_by_machine_id, rm_conf):
```

Comment:
In an ideal world… 🤷🏻‍♂️ But no problem, let’s keep it.

```suggestion
def test_canonical_facts_request(get_proxies, fetch_system_by_machine_id, get_canonical_facts, post, rm_conf):
```